OpenChart/src/UI/NoteField/BeatLines.cs

```diff
@@ -0,0 +1,117 @@
+using Gtk;
+using System;
+using System.Collections.Generic;
+
+namespace OpenChart.UI.NoteField
+{
+    /// <summary>
+    /// Note field widget for displaying beat lines.
+    /// </summary>
+    public class BeatLines : IWidget
+    {
+        /// <summary>
+        /// The drawing settings to use for the beat lines.
+        /// </summary>
+        public BeatLineSettings BeatLineSettings { get; private set; }
+
+        /// <summary>
+        /// The settings for the note field.
+        /// </summary>
+        public NoteFieldSettings NoteFieldSettings { get; private set; }
+
+        DrawingArea drawingArea;
+        public Widget GetWidget() => drawingArea;
+
+        /// <summary>
+        /// Creates a new BeatLines instance.
+        /// </summary>
+        /// <param name="noteFieldSettings">The settings for the note field.</param>
+        /// <param name="beatLineSettings">The settings for the BeatLines instance.</param>
+        public BeatLines(NoteFieldSettings noteFieldSettings, BeatLineSettings beatLineSettings)
+        {
+            BeatLineSettings = beatLineSettings;
+            NoteFieldSettings = noteFieldSettings;
+
+            drawingArea = new DrawingArea();
+            drawingArea.Drawn += onDrawn;
+
+            // Resize the widget when the note field is updated.
+            NoteFieldSettings.ChartEventBus.Anything += delegate { resizeToFit(); };
+            resizeToFit();
+        }
+
+        /// <summary>
+        /// Draws horizontal beat lines at the given positions.
+        /// </summary>
+        private void drawBeatLines(Cairo.Context cr, int width, Color color, int thickness, List<int> positions)
+        {
+            cr.SetSourceColor(color.AsCairoColor());
+            cr.LineWidth = thickness;
+
+            // Fix the line not being the right thickness if the thickness is an odd number.
+            var applyOffset = (thickness % 2 == 1);
+
+            foreach (var y in positions)
+            {
+                var offsetY = y + (applyOffset ? 0.5 : 0);
```

Comment:
maybe move the applyOffset logic to the variable? the resulting code could look something like this:

```cs
var offset = ((thickness % 2 == 1) ? 0.5 : 0)
foreach (var y in positions)
{
    var offsetY = y + offset;
    [...]
}
```

Reply:
That's a 2AM moment for sure

Reply:
mood